ports/samd/pin_af.c

```diff
@@ -62,45 +73,49 @@ const machine_pin_obj_t *pin_find(mp_obj_t pin) {
     }
     if (mp_obj_is_small_int(pin)) {
         // Pin defined by pin number for PAnn, PBnn, etc.
-        self = get_pin_obj_ptr(mp_obj_get_int(pin));
-    } else if (mp_obj_is_str(pin)) {
+        return get_pin_obj_ptr(mp_obj_get_int(pin));
+    }
+    #if MICROPY_HW_PIN_BOARD_CPU
+    // See if the pin name matches a board pin
+    self = pin_find_named_pin(&machine_pin_board_pins_locals_dict, pin);
+
```

Comment:
It might be a little cleaner to write:
```c
if (self != NULL) {
    return self;
}
```
And use the same logic just below for searching CPU pins.  Then at the end of the function just `mp_raise_ValueError(MP_ERROR_TEXT("not a Pin"));` unconditionally, because if you get there then there is no matching pin (this is how stm32's `pin_find` works).

Reply:
That's indeed the way it's done in other ports. Changed.